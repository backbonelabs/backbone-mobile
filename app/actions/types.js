// app actions
export const SET_CONFIG = 'SET_CONFIG';
export const SHOW_FULL_MODAL = 'SHOW_FULL_MODAL';
export const HIDE_FULL_MODAL = 'HIDE_FULL_MODAL';
export const SHOW_PARTIAL_MODAL = 'SHOW_PARTIAL_MODAL';
export const HIDE_PARTIAL_MODAL = 'HIDE_PARTIAL_MODAL';
export const UPDATE_BLUETOOTH_STATE = 'UPDATE_BLUETOOTH_STATE';
export const SET_TITLE_BAR = 'SET_TITLE_BAR';

// auth actions
export const LOGIN = 'LOGIN';
export const LOGIN__START = 'LOGIN__START';
export const LOGIN__ERROR = 'LOGIN__ERROR';
export const SIGNUP = 'SIGNUP';
export const SIGNUP__START = 'SIGNUP__START';
export const SIGNUP__ERROR = 'SIGNUP__ERROR';
export const PASSWORD_RESET = 'PASSWORD_RESET';
export const PASSWORD_RESET__START = 'PASSWORD_RESET__START';
export const PASSWORD_RESET__ERROR = 'PASSWORD_RESET__ERROR';
export const SIGN_OUT = 'SIGN_OUT';
export const SIGN_OUT__START = 'SIGN_OUT__START';
export const SIGN_OUT__ERROR = 'SIGN_OUT__ERROR';
export const SET_ACCESS_TOKEN = 'SET_ACCESS_TOKEN';
export const SET_ACCESS_TOKEN__START = 'SET_ACCESS_TOKEN__START';
export const SET_ACCESS_TOKEN__ERROR = 'SET_ACCESS_TOKEN__ERROR';

// dashboard actions
export const SELECT_LEVEL = 'SELECT_LEVEL';
export const SELECT_SESSION = 'SELECT_SESSION';

// device actions
export const DEVICE_CONNECT = 'DEVICE_CONNECT';
export const DEVICE_CONNECT__START = 'DEVICE_CONNECT__START';
export const DEVICE_CONNECT__ERROR = 'DEVICE_CONNECT__ERROR';
export const DEVICE_CONNECT_STATUS = 'DEVICE_CONNECT_STATUS';
export const DEVICE_CONNECT_STATUS__START = 'DEVICE_CONNECT_STATUS__START';
export const DEVICE_CONNECT_STATUS__ERROR = 'DEVICE_CONNECT_STATUS__ERROR';
export const DEVICE_DISCONNECT = 'DEVICE_DISCONNECT';
export const DEVICE_DISCONNECT__START = 'DEVICE_DISCONNECT__START';
export const DEVICE_DISCONNECT__ERROR = 'DEVICE_DISCONNECT__ERROR';
export const DEVICE_FORGET = 'DEVICE_FORGET';
export const DEVICE_FORGET__START = 'DEVICE_FORGET__START';
export const DEVICE_FORGET__ERROR = 'DEVICE_FORGET__ERROR';
export const DEVICE_GET_INFO = 'DEVICE_GET_INFO';
export const DEVICE_GET_INFO__START = 'DEVICE_GET_INFO__START';
export const DEVICE_GET_INFO__ERROR = 'DEVICE_GET_INFO__ERROR';
export const DEVICE_SELF_TEST_REQUESTED = 'DEVICE_SELF_TEST_REQUESTED';
export const DEVICE_SELF_TEST_UPDATED = 'DEVICE_SELF_TEST_UPDATED';
export const DEVICE_RESTORE_SAVED_SESSION = 'DEVICE_RESTORE_SAVED_SESSION';
export const DEVICE_CLEAR_SAVED_SESSION = 'DEVICE_CLEAR_SAVED_SESSION';
export const DEVICE_FIRMWARE_UPDATE_STARTED = 'DEVICE_FIRMWARE_UPDATE_STARTED';
export const DEVICE_FIRMWARE_UPDATE_ENDED = 'DEVICE_FIRMWARE_UPDATE_ENDED';
export const DEVICE_SET_PENDING_UPDATE = 'DEVICE_SET_PENDING_UPDATE';
export const DEVICE_UNSET_PENDING_UPDATE = 'DEVICE_UNSET_PENDING_UPDATE';

// posture actions
export const SET_SESSION_TIME = 'SET_SESSION_TIME';
export const SET_SESSION_TIME__START = 'SET_SESSION_TIME__START';
export const SET_SESSION_TIME__ERROR = 'SET_SESSION_TIME__ERROR';

// support actions
export const UPDATE_SUPPORT_MESSAGE = 'UPDATE_SUPPORT_MESSAGE';
export const CREATE_SUPPORT_TICKET = 'CREATE_SUPPORT_TICKET';
export const CREATE_SUPPORT_TICKET__START = 'CREATE_SUPPORT_TICKET__START';
export const CREATE_SUPPORT_TICKET__ERROR = 'CREATE_SUPPORT_TICKET__ERROR';

// user actions
export const FETCH_USER = 'FETCH_USER';
export const FETCH_USER__START = 'FETCH_USER__START';
export const FETCH_USER__ERROR = 'FETCH_USER__ERROR';
export const UPDATE_USER = 'UPDATE_USER';
export const UPDATE_USER__START = 'UPDATE_USER__START';
export const UPDATE_USER__ERROR = 'UPDATE_USER__ERROR';
export const UPDATE_USER_SETTINGS = 'UPDATE_USER_SETTINGS';
export const UPDATE_USER_SETTINGS__START = 'UPDATE_USER_SETTINGS__START';
export const UPDATE_USER_SETTINGS__ERROR = 'UPDATE_USER_SETTINGS__ERROR';
export const FETCH_USER_SESSIONS = 'FETCH_USER_SESSIONS';
export const FETCH_USER_SESSIONS__START = 'FETCH_USER_SESSIONS__START';
export const FETCH_USER_SESSIONS__ERROR = 'FETCH_USER_SESSIONS__ERROR';
export const PREPARE_USER_UPDATE = 'PREPARE_USER_UPDATE';
export const PREPARE_USER_UPDATE__START = 'PREPARE_USER_UPDATE__START';
export const PREPARE_USER_UPDATE__ERROR = 'PREPARE_USER_UPDATE__ERROR';
export const FETCH_USER_WORKOUTS = 'FETCH_USER_WORKOUTS';
export const FETCH_USER_WORKOUTS__START = 'FETCH_USER_WORKOUTS__START';
export const FETCH_USER_WORKOUTS__ERROR = 'FETCH_USER_WORKOUTS__ERROR';
