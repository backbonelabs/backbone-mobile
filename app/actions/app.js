import {
  SET_CONFIG,
  SET_TITLE_BAR,
  SHOW_FULL_MODAL,
  HIDE_FULL_MODAL,
  SHOW_PARTIAL_MODAL,
  HIDE_PARTIAL_MODAL,
  SET_NEXT_ROUTE,
  TOGGLE_OVERRIDE_BACK_BUTTON,
} from './types';

export default {
  setConfig(config) {
    return {
      type: SET_CONFIG,
      payload: config,
    };
  },
  setTitleBar(titleBarConfig) {
    return {
      type: SET_TITLE_BAR,
      payload: titleBarConfig,
    };
  },
  showFullModal(modalConfig) {
    return {
      type: SHOW_FULL_MODAL,
      payload: modalConfig,
    };
  },
  hideFullModal() {
    return { type: HIDE_FULL_MODAL };
  },
  showPartialModal(modalConfig) {
    return {
      type: SHOW_PARTIAL_MODAL,
      payload: modalConfig,
    };
  },
  hidePartialModal() {
    return { type: HIDE_PARTIAL_MODAL };
  },
  setNextRoute(route) {
    return {
      type: SET_NEXT_ROUTE,
      payload: route,
    };
  },
  toggleOverrideBackButton(state) {
    return {
      type: TOGGLE_OVERRIDE_BACK_BUTTON,
      payload: state,
    };
  },
};
