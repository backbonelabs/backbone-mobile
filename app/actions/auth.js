import { NativeModules } from 'react-native';
import Fetcher from '../utils/Fetcher';
import SensitiveInfo from '../utils/SensitiveInfo';
import constants from '../utils/constants';

const { Environment, Mixpanel } = NativeModules;

const loginStart = () => ({ type: 'LOGIN__START' });

const login = payload => ({
  type: 'LOGIN',
  payload,
});

const loginError = error => ({
  type: 'LOGIN__ERROR',
  payload: error,
  error: true,
});

const signupStart = () => ({ type: 'SIGNUP__START' });

const signup = payload => ({
  type: 'SIGNUP',
  payload,
});

const signOut = () => ({ type: 'SIGN__OUT' });

const signupError = error => ({
  type: 'SIGNUP__ERROR',
  payload: error,
  error: true,
});

const passwordResetStart = () => ({ type: 'PASSWORD_RESET__START' });

const passwordReset = payload => ({
  type: 'PASSWORD_RESET',
  payload,
});

const passwordResetError = error => ({
  type: 'PASSWORD_RESET__ERROR',
  payload: error,
  error: true,
});

const setAccessToken = token => ({
  type: 'SET_ACCESS_TOKEN',
  payload: token,
});

export default {
  login(user) {
    return (dispatch) => {
      dispatch(loginStart());

      return Fetcher.post({
        url: `${Environment.API_SERVER_URL}/auth/login`,
        body: JSON.stringify(user),
      })
        .then(response => response.json()
          .then((body) => {
            if (body.error) {
              // Error received from API server
              dispatch(loginError(
                new Error(body.error)
              ));
            } else {
              // Identify user for Mixpanel tracking
              Mixpanel.identify(body._id);
              Mixpanel.set({ $email: body.email });
              dispatch(login(body));
            }
          })
        )
        .catch(() => {
          // Network error
          dispatch(loginError(
            new Error('We are encountering server issues. Please try again later.')
          ));
        });
    };
  },

  signup(user) {
    return (dispatch) => {
      dispatch(signupStart());

      return Fetcher.post({
        url: `${Environment.API_SERVER_URL}/users/`,
        body: JSON.stringify(user),
      })
        .then((response) => response.json()
          .then(body => {
            // Error received from API server
            if (body.error) {
              dispatch(signupError(
                new Error(body.error)
              ));
            } else {
              // Identify user for Mixpanel tracking
              Mixpanel.identify(body.user._id);
              Mixpanel.set({ $email: body.user.email });
              Mixpanel.track('signup');
              dispatch(signup(body));
            }
          })
        )
        .catch(() => (
          // Network error
          dispatch(signupError(
            new Error('We are encountering server issues. Please try again later.')
          ))
        ));
    };
  },

  reset(user) {
    return (dispatch) => {
      dispatch(passwordResetStart());

      return Fetcher.post({
        url: `${Environment.API_SERVER_URL}/auth/password-reset-token`,
        body: JSON.stringify(user),
      })
        .then((response) => {
          if (response.ok) {
            dispatch(passwordReset(response.ok));
          } else {
            return response.json()
              .then(body => (
                dispatch(passwordResetError(
                  new Error(body.error)
                ))
              ));
          }
        })
        .catch(() => (
          // Network error
          dispatch(passwordResetError(
            new Error('We are encountering server issues. Please try again later.')
          ))
        ));
    };
  },
  signOut() {
    return (dispatch) => {
      SensitiveInfo.deleteItem(constants.accessTokenStorageKey);
      SensitiveInfo.deleteItem(constants.userStorageKey);
      dispatch(signOut());
    };
  },
  setAccessToken,
};
