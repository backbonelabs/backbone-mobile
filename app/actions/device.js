import {
  NativeModules,
  NativeEventEmitter,
} from 'react-native';
import Fetcher from '../utils/Fetcher';
import constants from '../utils/constants';
import Mixpanel from '../utils/Mixpanel';
import SensitiveInfo from '../utils/SensitiveInfo';

const {
  Environment,
  DeviceManagementService,
  DeviceInformationService,
} = NativeModules;
const { storageKeys } = constants;
const firmwareUrl = `${Environment.API_SERVER_URL}/firmware`;
const deviceManagementServiceEvents = new NativeEventEmitter(DeviceManagementService);

const connectStart = () => ({ type: 'DEVICE_CONNECT__START' });

const connect = payload => ({
  type: 'DEVICE_CONNECT',
  payload,
});

const connectError = payload => ({
  type: 'DEVICE_CONNECT__ERROR',
  payload,
});

function setConnectEventListener(dispatch, getInfo) {
  const connectionStatusListener = deviceManagementServiceEvents.addListener(
    'ConnectionStatus',
    status => {
      if (status.message) {
        dispatch(connectError(status));
        Mixpanel.trackError({
          message: status.message,
          path: 'app/actions/device',
          stackTrace: ['setConnectEventListener', 'deviceManagementServiceEvents.addListener'],
        });
      } else {
        dispatch(connect(status));
        // Call getInfo to fetch latest device information
        dispatch(getInfo());
      }
      connectionStatusListener.remove();
    }
  );
}

function checkFirmware(firmwareVersion) {
  // Fetch device firmware details
  return Fetcher.get({ url: firmwareUrl })
    .then(res => res.json()
      .then(body => {
        let updateAvailable = false;
        // Split firmware versions into array for digit-by-digit comparison
        const newFirmware = body.version.split('.');
        const currentFirmware = firmwareVersion.split('.');

        // Check and compare each of the firmware version digits in order
        // Use for loop, since forEach can't be interrupted
        for (let i = 0; i < newFirmware.length; i++) {
          if (parseInt(newFirmware[i], 10) > parseInt(currentFirmware[i], 10)) {
            updateAvailable = true;
            break;
          }
        }

        return updateAvailable;
      })
    );
}

const disconnectStart = () => ({ type: 'DEVICE_DISCONNECT__START' });

const disconnect = () => ({
  type: 'DEVICE_DISCONNECT',
});

const disconnectError = error => ({
  type: 'DEVICE_DISCONNECT__ERROR',
  payload: error,
});

const forgetStart = () => ({ type: 'DEVICE_FORGET__START' });

const forget = () => ({
  type: 'DEVICE_FORGET',
});

const forgetError = error => ({
  type: 'DEVICE_FORGET__ERROR',
  payload: error,
});

const getInfoStart = () => ({ type: 'DEVICE_GET_INFO__START' });

const getInfo = payload => ({
  type: 'DEVICE_GET_INFO',
  payload,
});

const getInfoError = error => ({
  type: 'DEVICE_GET_INFO__ERROR',
  payload: error,
});

const deviceActions = {
  connect(deviceIdentifier) {
    return (dispatch) => {
      dispatch(connectStart());
      setConnectEventListener(dispatch, deviceActions.getInfo);
      // Connect to device with specified identifier
      DeviceManagementService.connectToDevice(deviceIdentifier);
    };
  },
  disconnect() {
    return (dispatch, getState) => {
      // Attempt disconnect only if device is connected
      if (getState().device.isConnected) {
        dispatch(disconnectStart());
        DeviceManagementService.cancelConnection(err => {
          if (err) {
            dispatch(disconnectError(err));
            Mixpanel.trackError({
              message: err.message,
              path: 'app/actions/device',
              stackTrace: ['deviceActions.disconnect', 'DeviceManagementService.cancelConnection'],
            });
          } else {
            dispatch(disconnect());
          }
        });
      }
    };
  },
  forget() {
    return (dispatch) => {
      dispatch(forgetStart());
      // Disconnect device before attempting to forget
      DeviceManagementService.cancelConnection(err => {
        if (err) {
          dispatch(forgetError(err));
          Mixpanel.trackError({
            message: err.message,
            path: 'app/actions/device',
            stackTrace: ['deviceActions.forget', 'DeviceManagementService.cancelConnection'],
          });
        } else {
          // Remove device information from local storage
          SensitiveInfo.deleteItem(storageKeys.DEVICE);
          dispatch(forget());
        }
      });
    };
  },
  getInfo() {
    return (dispatch, getState) => {
      const { device: deviceState } = getState();
      dispatch(getInfoStart());

      // Get latest information if device is connected
      if (deviceState.isConnected) {
        DeviceInformationService.getDeviceInformation((err, results) => {
          if (err) {
            dispatch(getInfoError(err));
            Mixpanel.trackError({
              message: err.message,
              path: 'app/actions/device',
              stackTrace: ['deviceActions.getInfo', 'DeviceManagementService.getDeviceInformation'],
            });
          } else {
            // If there's new firmware, set updateAvailable to true
            checkFirmware(results.firmwareVersion)
              .then(updateAvailable => {
                // Clone device in order to add updateAvailable property
                const resultsClone = { ...results, updateAvailable };

                // Store device information in local storage
                SensitiveInfo.setItem(storageKeys.DEVICE, resultsClone);
                dispatch(getInfo(resultsClone));
              });
          }
        });
      } else {
        // Get locally stored device information
        SensitiveInfo.getItem(storageKeys.DEVICE)
          .then(device => {
            if (device) {
              checkFirmware(device.firmwareVersion)
                .then(updateAvailable => {
                  // Clone device in order to add updateAvailable property
                  const deviceClone = { ...device, updateAvailable };

                  // Update device store
                  dispatch(getInfo(deviceClone));
                });
              // Attempt to connect to locally stored device
              dispatch(deviceActions.connect(device.identifier));
            } else {
              // No locally stored device, set to empty object
              dispatch(getInfo({}));
            }
          });
      }
    };
  },
};

export default deviceActions;
