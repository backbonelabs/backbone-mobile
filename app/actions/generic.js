import {
  NativeModules,
  NativeAppEventEmitter,
} from 'react-native';

const { DeviceManagementService } = NativeModules;

const connectStart = () => ({ type: 'CONNECT__START' });

const connect = payload => ({
  type: 'CONNECT',
  payload,
});

const connectError = error => ({
  type: 'CONNECT__ERROR',
  error,
});

const scanStart = () => ({ type: 'SCAN__START' });

const scan = payload => ({
  type: 'SCAN',
  payload,
});

const scanError = error => ({
  type: 'SCAN__ERROR',
  error,
});

const forgetStart = () => ({ type: 'FORGET__START' });

const forget = payload => ({
  type: 'FORGET',
  payload,
});

const forgetError = payload => ({
  type: 'FORGET__ERROR',
  payload,
});

async function ConnectEventListener(dispatch) {
  let eventListener = await NativeAppEventEmitter.once('ConnectionStatus', status => {
    dispatch(connect(status));
  });

  return eventListener;
}

async function ScanEventListener(dispatch) {
  let eventListener = await NativeAppEventEmitter.once('DevicesFound', deviceList => {
    dispatch(scan(deviceList));
  });

  return eventListener;
}

export default {
  connect() {
    return (dispatch) => {
      dispatch(connectStart());

      return ConnectEventListener(dispatch)
        .then(() => DeviceManagementService.connectToDevice());
    };
  },
  scan() {
    return (dispatch) => {
      dispatch(scanStart());

      return ScanEventListener(dispatch)
        .then(() => DeviceManagementService.scanForDevices());
    };
  },
  forget() {
    return (dispatch) => {
      dispatch(forgetStart());

      return DeviceManagementService.forgetDevice()
        .then(() => dispatch(forget()))
        .catch((error) => dispatch(forgetError(error)));
    };
  },
};
