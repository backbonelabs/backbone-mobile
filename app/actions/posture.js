import { SET_SESSION_PARAMETERS } from './types';

export default {
  setSessionParameters(param) {
    return {
      type: SET_SESSION_PARAMETERS,
      payload: param,
    };
  },
};
