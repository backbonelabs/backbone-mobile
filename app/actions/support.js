import { NativeModules } from 'react-native';
import {
  CREATE_SUPPORT_TICKET,
  UPDATE_SUPPORT_MESSAGE,
} from './types';
import store from '../store';
import constants from '../utils/constants';
import Fetcher from '../utils/Fetcher';
import Mixpanel from '../utils/Mixpanel';

const { Environment } = NativeModules;
const { errorMessages } = constants;

const handleNetworkError = mixpanelEvent => {
  Mixpanel.track(`${mixpanelEvent}-serverError`);
  throw new Error(errorMessages.NETWORK_ERROR);
};

export default {
  updateMessage(message) {
    return {
      type: UPDATE_SUPPORT_MESSAGE,
      payload: message,
    };
  },
  createTicket(message) {
    const { auth: { accessToken }, user: { user } } = store.getState();
    const createTicketEventName = 'createTicket';

    return {
      type: CREATE_SUPPORT_TICKET,
      payload: () => Fetcher.post({
        url: `${Environment.API_SERVER_URL}/support`,
        headers: { Authorization: `Bearer ${accessToken}` },
        body: JSON.stringify({
          _id: user._id,
          message,
        }),
      })
        .catch(() => handleNetworkError(createTicketEventName))
        .then(response => response.json())
        .then((body) => {
          if (body.error) {
            Mixpanel.trackWithProperties(`${createTicketEventName}-error`, {
              errorMessage: body.error,
            });

            throw new Error(body.error);
          }
          Mixpanel.track(`${createTicketEventName}-success`);
          return;
        }),
    };
  },
};
