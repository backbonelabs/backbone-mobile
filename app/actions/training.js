import {
  SELECT_LEVEL,
} from './types';

export default {
  selectLevel(level) {
    return {
      type: SELECT_LEVEL,
      payload: level,
    };
  },
};
