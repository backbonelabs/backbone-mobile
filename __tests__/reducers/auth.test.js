import * as types from '../../app/actions/types';
import authReducer from '../../app/reducers/auth';

describe('__Auth Reducer__', () => {
  const initialState = {
    accessToken: null,
    passwordResetSent: false,
    inProgress: false,
    errorMessage: null,
    userId: null,
  };

  const user = {
    _id: 'testId',
    accessToken: 'testToken',
    email: 'test@mail.com',
    nickname: 'testNick',
    password: 'password',
  };

  const error = {
    message: 'testError',
  };

  test('should return initial state', () => {
    expect(authReducer(initialState, {})).toMatchSnapshot();
  });

  test('should handle LOGIN', () => {
    expect(authReducer(initialState, {
      type: types.LOGIN,
      payload: user,
    })).toMatchSnapshot();
  });

  test('should handle LOGIN__START', () => {
    expect(authReducer(initialState, {
      type: types.LOGIN__START,
    })).toMatchSnapshot();
  });

  test('should handle LOGIN__ERROR', () => {
    expect(authReducer(initialState, {
      type: types.LOGIN__ERROR,
      payload: error,
    })).toMatchSnapshot();
  });

  test('should handle SIGNUP', () => {
    const { accessToken, ...userObj } = user;
    expect(authReducer(initialState, {
      type: types.SIGNUP,
      payload: { accessToken, user: userObj },
    })).toMatchSnapshot();
  });

  test('should handle SIGNUP__START', () => {
    expect(authReducer(initialState, {
      type: types.SIGNUP__START,
    })).toMatchSnapshot();
  });

  test('should handle SIGNUP__ERROR', () => {
    expect(authReducer(initialState, {
      type: types.SIGNUP__ERROR,
      payload: error,
    })).toMatchSnapshot();
  });

  test('should handle PASSWORD_RESET', () => {
    expect(authReducer(initialState, {
      type: types.PASSWORD_RESET,
      payload: true,
    })).toMatchSnapshot();
  });

  test('should handle PASSWORD_RESET__START', () => {
    expect(authReducer(initialState, {
      type: types.PASSWORD_RESET__START,
    })).toMatchSnapshot();
  });

  test('should handle PASSWORD_RESET__ERROR', () => {
    expect(authReducer(initialState, {
      type: types.PASSWORD_RESET__ERROR,
      payload: error,
    })).toMatchSnapshot();
  });

  test('should handle SIGN_OUT', () => {
    expect(authReducer(initialState, {
      type: types.SIGN_OUT,
    })).toMatchSnapshot();
  });

  test('should handle SET_ACCESS_TOKEN', () => {
    expect(authReducer(initialState, {
      type: types.SET_ACCESS_TOKEN,
    })).toMatchSnapshot();
  });
});
