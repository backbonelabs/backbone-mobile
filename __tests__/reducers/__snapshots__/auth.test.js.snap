// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__Auth Reducer__ should handle LOGIN 1`] = `
Object {
  "accessToken": "testToken",
  "errorMessage": null,
  "inProgress": false,
  "passwordResetSent": false,
  "userId": "testId",
}
`;

exports[`__Auth Reducer__ should handle LOGIN__ERROR 1`] = `
Object {
  "accessToken": null,
  "errorMessage": "testError",
  "inProgress": false,
  "passwordResetSent": false,
  "userId": null,
}
`;

exports[`__Auth Reducer__ should handle LOGIN__START 1`] = `
Object {
  "accessToken": null,
  "errorMessage": null,
  "inProgress": true,
  "passwordResetSent": false,
  "userId": null,
}
`;

exports[`__Auth Reducer__ should handle PASSWORD_RESET 1`] = `
Object {
  "accessToken": null,
  "errorMessage": null,
  "inProgress": false,
  "passwordResetSent": true,
  "userId": null,
}
`;

exports[`__Auth Reducer__ should handle PASSWORD_RESET__ERROR 1`] = `
Object {
  "accessToken": null,
  "errorMessage": "testError",
  "inProgress": false,
  "passwordResetSent": false,
  "userId": null,
}
`;

exports[`__Auth Reducer__ should handle PASSWORD_RESET__START 1`] = `
Object {
  "accessToken": null,
  "errorMessage": null,
  "inProgress": true,
  "passwordResetSent": false,
  "userId": null,
}
`;

exports[`__Auth Reducer__ should handle SET_ACCESS_TOKEN 1`] = `
Object {
  "accessToken": undefined,
  "errorMessage": null,
  "inProgress": false,
  "passwordResetSent": false,
  "userId": null,
}
`;

exports[`__Auth Reducer__ should handle SIGN_OUT 1`] = `
Object {
  "accessToken": null,
  "errorMessage": null,
  "inProgress": false,
  "passwordResetSent": false,
  "userId": null,
}
`;

exports[`__Auth Reducer__ should handle SIGNUP 1`] = `
Object {
  "accessToken": "testToken",
  "errorMessage": null,
  "inProgress": false,
  "passwordResetSent": false,
  "userId": "testId",
}
`;

exports[`__Auth Reducer__ should handle SIGNUP__ERROR 1`] = `
Object {
  "accessToken": null,
  "errorMessage": "testError",
  "inProgress": false,
  "passwordResetSent": false,
  "userId": null,
}
`;

exports[`__Auth Reducer__ should handle SIGNUP__START 1`] = `
Object {
  "accessToken": null,
  "errorMessage": null,
  "inProgress": true,
  "passwordResetSent": false,
  "userId": null,
}
`;

exports[`__Auth Reducer__ should return initial state 1`] = `
Object {
  "accessToken": null,
  "errorMessage": null,
  "inProgress": false,
  "passwordResetSent": false,
  "userId": null,
}
`;
